include/swift/SIL/SILInstruction.h

```diff
@@ -693,6 +745,184 @@ inline SingleValueInstruction *SILNode::castToSingleValueInstruction() {
            inst->getKind() <= SILInstructionKind::Last_##ID;    \
   }
 
+/// A value base result of a multiple value instruction.
+///
+/// *NOTE* We want this to be a pure abstract class that does not add /any/ size
+/// to subclasses.
+class MultipleValueInstructionResult : public ValueBase {
+public:
+  /// Create a new multiple value instruction result.
+  ///
+  /// \arg subclassDeltaOffset This is the delta offset in our parent object's
+  /// layout in between the end of the MultipleValueInstruction object and the
+  /// end of the specific subclass object.
+  ///
+  /// *NOTE* subclassDeltaOffset must be use only 5 bits. This gives us to
+  /// support subclasses up to 32 bytes in size. We can scavange up to 6 more
+  /// bits from ValueBase if this is not large enough.
+  MultipleValueInstructionResult(ValueKind valueKind, unsigned index,
+                                 SILType type,
+                                 ValueOwnershipKind ownershipKind);
+
+  /// Return the parent instruction of this result.
+  MultipleValueInstruction *getParent();
+
+  const MultipleValueInstruction *getParent() const {
+    return const_cast<MultipleValueInstructionResult *>(this)->getParent();
+  }
+
+  unsigned getIndex() const;
+
+  /// Get the ownership kind assigned to this result by its parent.
+  ///
+  /// This is stored in the bottom 3 bits of ValueBase's subclass data.
+  ValueOwnershipKind getOwnershipKind() const;
+
+  SILNode *getCanonicalSILNodeInObject();
+  const SILNode *getCanonicalSILNodeInObject() const;
```

Comment:
I don't know why you have these overrides in subclasses, but they have the wrong name now.